src/msd_lite_stat_json.c

```diff
@@ -0,0 +1,493 @@
+
+	return (error);
+}
+
+static void
+gen_hub_stat_json_entry_enum_cb(tpt_p tpt, str_hub_p str_hub, void *udata) {
+	json_enum_ctx_p ctx = udata;
+	http_srv_cli_p cli = ctx->cli;
+	io_buf_p buf = http_srv_cli_get_buf(cli);
+	str_hub_cli_p strh_cli, strh_cli_temp;
+	time_t cur_time, time_conn;
+	char straddr[STR_ADDR_LEN], straddr2[STR_ADDR_LEN], ifname[(IFNAMSIZ + 1)], str_time[64];
+	str_src_conn_mc_p conn_mc;
+	int first_cli = 1;
+
+	cur_time = gettime_monotonic();
+
+	if (ctx->first) {
+		ctx->first = 0;
+	} else {
+		io_buf_printf(buf, ",\n");
+	}
+
+	io_buf_printf(buf, "    {\n");
+	io_buf_printf(buf, "      \"name\": \"");
+	json_escape_str(buf, str_hub->name);
+	io_buf_printf(buf, "\",\n");
+	
+	io_buf_printf(buf, "      \"rate_in\": %"PRIu64",\n", str_hub->baud_rate_in);
+	io_buf_printf(buf, "      \"dropped_count\": %"PRIu64",\n", str_hub->dropped_count);
+	
+	/* Source Info */
+	conn_mc = &str_hub->src_conn_params.mc;
+	if (0 != sa_addr_port_to_str(&conn_mc->udp.addr, straddr,
+	    sizeof(straddr), NULL)) {
+		memcpy(straddr, "unknown", 8);
+	}
+	ifname[0] = 0;
+	if_indextoname(conn_mc->if_index, ifname);
+	
+	io_buf_printf(buf, "      \"source\": \"%s@%s\",\n", straddr, ifname);
+
+	/* Clients */
+	io_buf_printf(buf, "      \"clients\": [\n");
+	
+	TAILQ_FOREACH_SAFE(strh_cli, &str_hub->cli_head, next, strh_cli_temp) {
+		if (!first_cli) {
+			io_buf_printf(buf, ",\n");
+		}
+		first_cli = 0;
+
+		if (0 != sa_addr_port_to_str(&strh_cli->remonte_addr,
+		    straddr, sizeof(straddr), NULL)) {
+			memcpy(straddr, "unknown", 8);
+		}
+		
+		time_conn = (cur_time - strh_cli->conn_time);
+		fmt_as_uptime(&time_conn, str_time, sizeof(str_time));
+
+		io_buf_printf(buf, "        {\n");
+		io_buf_printf(buf, "          \"ip\": \"%s\",\n", straddr);
+		io_buf_printf(buf, "          \"agent\": \"");
+		if (strh_cli->user_agent) {
+			json_escape_str(buf, (char*)strh_cli->user_agent);
+		}
+		io_buf_printf(buf, "\",\n");
+		io_buf_printf(buf, "          \"time\": \"%s\"\n", str_time);
+		io_buf_printf(buf, "        }");
+	}
+	
+	io_buf_printf(buf, "\n      ]\n"); // End clients
+	io_buf_printf(buf, "    }"); // End hub
+}
+
+static void
+gen_hub_stat_json_enum_done_cb(tpt_p tpt __unused, size_t send_msg_cnt __unused,
+    size_t error_cnt, void *udata) {
+	json_enum_ctx_p ctx = udata;
+	http_srv_cli_p cli = ctx->cli;
+	io_buf_p buf = http_srv_cli_get_buf(cli);
+	http_srv_resp_p	resp = http_srv_cli_get_resp(cli);
+	static const char *cttype = 	"Content-Type: application/json\r\n"
+					"Pragma: no-cache";
+
+	/* Close JSON */
+	io_buf_printf(buf, "\n  ]\n}\n");
+
+	free(ctx);
+
+	if (0 == error_cnt) {
+		resp->status_code = 200;
+		resp->p_flags |= HTTP_SRV_RESP_P_F_CONTENT_LEN;
+		resp->hdrs_count = 1;
+		resp->hdrs[0].iov_base = MK_RW_PTR(cttype);
+		resp->hdrs[0].iov_len = 48;
+	} else {
+		resp->status_code = 500;
+	}
+	http_srv_resume_responce(cli);
+}
+
+/* Admin Page HTML */
+static const char *admin_html = 
```

Comment:
It is too big to be inside ```.c``` file, and it is hard to edit ```.html``` here.

https://tratt.net/laurie/blog/2022/whats_the_most_portable_way_to_include_binary_blobs_in_an_executable.html
I like: "Assembler tricks" and "Preprocessing" with ```hexdump``` or ```xxd```. 
Also please make sure that you will use ```sizeof()``` or precalculated size instead of ```strlen()```.